JoinColony/colonyCDapp
# Fail_all_ pending txs (including `CREATED` and `READY`)

## Description

This PR adds transactions with the status `CREATED` and `READY` to the ones that are being cleared when the user refreshes the page.

I know this is not the ideal approach (we would like to retry those transactions), but with the current transaction API that's all we can do for now.

To fix the current issues on QA and production we will have to write a migration script that fails all of these already existing transactions.

## Testing

This is quite hard to test locally, so you will have to create a transaction for your user manually:

1) Create unfinished transactions manually:

```gql
mutation CreateUnfinishedTransactions {
  tx1: createTransaction(
    input: {id: "zBtVbwKJvFhxKbJSUTbma1-createColonyForFrontend", context: NetworkClient, from: "0xb77D57F4959eAfA0339424b83FcFaf9c15407461", colonyAddress: "0x52fC5a44C7f53C42f91BB20c2C9304F046481e91", groupId: "zBtVbwKJvFhxKbJSUTbmy", group: {id: "zBtVbwKJvFhxKbJSUTbma-createColonyForFrontend", groupId: "zBtVbwKJvFhxKbJSUTbmy", index: 10, key: ""}, methodName: "createColonyForFrontend", status: CREATED, blockHash: null, blockNumber: null, createdAt: "2024-08-26T14:22:03.445Z", hash: "0xa015a493bdc50ff0488e3522ea976e8c8be3e1239f3114199f0ec35b13a34088", identifier: "0x52fC5a44C7f53C42f91BB20c2C9304F046481e91", methodContext: null, options: "{}", params: "[\"0x0000000000000000000000000000000000000000\",\"Space Credits\",\"CREDS\",18,15,\"\",\"\"]", receipt: null, title: null, titleValues: null}
  ) {
    id
  }
  tx2:createTransaction(
    input: {id: "zBtVbwKJvFhxKbJSUTbmb1-createColonyForFrontend", context: NetworkClient, from: "0xb77D57F4959eAfA0339424b83FcFaf9c15407461", colonyAddress: "0x52fC5a44C7f53C42f91BB20c2C9304F046481e91", groupId: "zBtVbwKJvFhxKbJSUTbmy", group: {id: "zBtVbwKJvFhxKbJSUTbma-createColonyForFrontend", groupId: "zBtVbwKJvFhxKbJSUTbmy", index: 10, key: ""}, methodName: "createColonyForFrontend", status: READY, blockHash: null, blockNumber: null, createdAt: "2024-08-26T14:22:03.445Z", hash: "0xa015a493bdc50ff0488e3522ea976e8c8be3e1239f3114199f0ec35b13a34088", identifier: "0x52fC5a44C7f53C42f91BB20c2C9304F046481e91", methodContext: null, options: "{}", params: "[\"0x0000000000000000000000000000000000000000\",\"Space Credits\",\"CREDS\",18,15,\"\",\"\"]", receipt: null, title: null, titleValues: null}
  ) {
    id
  }
  tx3: createTransaction(
    input: {id: "zBtVbwKJvFhxKbJSUTbmc1-createColonyForFrontend", context: NetworkClient, from: "0xb77D57F4959eAfA0339424b83FcFaf9c15407461", colonyAddress: "0x52fC5a44C7f53C42f91BB20c2C9304F046481e91", groupId: "zBtVbwKJvFhxKbJSUTbmy", group: {id: "zBtVbwKJvFhxKbJSUTbma-createColonyForFrontend", groupId: "zBtVbwKJvFhxKbJSUTbmy", index: 10, key: ""}, methodName: "createColonyForFrontend", status: PENDING, blockHash: null, blockNumber: null, createdAt: "2024-08-26T14:22:03.445Z", hash: "0xa015a493bdc50ff0488e3522ea976e8c8be3e1239f3114199f0ec35b13a34088", identifier: "0x52fC5a44C7f53C42f91BB20c2C9304F046481e91", methodContext: null, options: "{}", params: "[\"0x0000000000000000000000000000000000000000\",\"Space Credits\",\"CREDS\",18,15,\"\",\"\"]", receipt: null, title: null, titleValues: null}
  ) {
    id
  }
}
```

2. Refresh the app using the address `0xb77D57F4959eAfA0339424b83FcFaf9c15407461` (Leela) and check if all transactions are either failed or succeeded.

## Diffs


**Changes** 🏗

* Also fail transactions that are `CREATED` or `READY` on user login

Resolves #2997
